src/Resources/app/storefront/src/js/filter-category-select.plugin.js

```diff
@@ -105,6 +122,120 @@ export default class FilterCategorySelectPlugin extends FilterBasePlugin {
         this.listing.changeListing();
     }
 
+    /**
+     * @param {HTMLObjectElement} iconShell
+     * @param {string} removeClass
+     * @param {string} addClass
+     * @public
+     */
+    toggleCategoryListIconState(iconShell, removeClass, addClass) {
+
+        if (typeof (iconShell) !== undefined && iconShell !== null) {
+            iconShell.classList.remove(removeClass);
+            iconShell.classList.add(addClass);
+        }
+    }
+
+    /**
+     * @param {HTMLObjectElement} checkbox
+     * @public
+     */
+    onChangeCheckBox(checkbox) {
+        const subCategories = checkbox.parentNode.getElementsByClassName('sub-item');
+        const arrow = checkbox.parentNode.querySelector('#arrow');
+        const subCategory = arrow !== null ? arrow.nextElementSibling : null;
+
+        if (!checkbox.checked) {
+            Iterator.iterate(subCategories, (category) => {
+                category.querySelector('.filter-category-select-checkbox').checked = false;
+            });
+            return;
+        }
+
+        const categoryNames = checkbox.value.split('_');
+
+        if (categoryNames.length > 0) {
+            let categoryNameAsId = ''
```

Comment:
Missing semicolon at the end.
```suggestion
            let categoryNameAsId = '';
```